dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -873,17 +873,11 @@ public Workflow getWorkflowByPath(@ApiParam(hidden = true) @Parameter(hidden = t
     }
 
     private Class<? extends Workflow> getSubClass(WorkflowSubClass subclass) {
-        final Class<? extends Workflow> targetClass;
-        if (subclass == WorkflowSubClass.SERVICE) {
-            targetClass = Service.class;
-        } else if (subclass == WorkflowSubClass.BIOWORKFLOW) {
-            targetClass = BioWorkflow.class;
-        } else if (subclass == WorkflowSubClass.APPTOOL) {
-            targetClass = AppTool.class;
-        } else {
+        try {
+            return subclass.getTargetClass();
+        } catch (Exception e) {
```

Comment:
You're not going to get an exception, unless `subclass` is `null`. And if `subclass`  is `null`, it will throw a `NullPointerException`, which this catch won't catch.

This is the nice thing about putting the logic in the enum;  you know it will always return a valid subclass.

So, your method should be, if it can ever be passed a null (can it? -- I don't know -- you'll have to look at the code):

```
if (subclass == null) {
 throw new CustomWebApplicationException("subclass is null blah blah"...
}
return subclass.getTargetClass();
```
If you know it can never be null, then all you need is

```
return subclass.getTargetClass();
```